### PR TITLE
Add JSON structure comparison step.

### PR DESCRIPTION
**Pull request by Ferada**
Same could be done for XML of course.

### CODE REVIEW

**Review by msackman — COMMENTED** [review]
It's a shame you have to provide the type twice - once in parseAs and once as the expected. Is there no way (via reflection maybe?) to create a fresh pointer to a value of the same type as expected, and thus be able to remove the parseAs argument?

**Comment by Ferada** [comment]
Yeah great point; looks like `reflect.New(reflect.TypeOf(a))` will do that.

File: http.go [hunk 1]
```diff
@@ -311,3 +314,25 @@ func (hc *HttpCall) ResponseBodyJSONSchema(schema string) Step {
...
+// marking the values that were actually present.
+func (hc *HttpCall) ResponseBodyJSONMatchesStruct(expected interface{}) Step {
+	return NewNamedStep("ResponseBodyJSONMatchesStruct", func() error {
+		parseAs := reflect.New(reflect.TypeOf(expected)).Interface()
+		if err := hc.ReceiveBody(); err != nil {
+			return err
+		} else if err := json.Unmarshal(hc.ResponseBody, parseAs); err != nil {
+			return err
+		} else if diff := pretty.Compare(parseAs, expected); diff != "" {
+			return fmt.Errorf("Did not match expected struct: (-got +want)\n%s", diff)
```
Comment (msackman) [inline http.go]: 'struct' is the wrong word - it may not be a struct. Maybe just 'value'?

**Review by msackman — COMMENTED** [review]
Excellent, just needs a merge in from master - apparently there's a conflict somewhere - I bet it's just in the imports though.

**Pull request merged**